### ISSUE TITLE
feat(external-link): persist URL check results to a file-based cache across CI runs

### ISSUE THREAD

**Issue by shinagawa-web** [shinagawa-web/gomarklint#270]
## Overview

gomarklint already deduplicates external link checks within a single run using an in-memory `sync.Map`. However, this cache is discarded when the process exits, so every CI run re-checks every URL from scratch.

Adding a file-based persistent cache allows results from a previous run to be reused, significantly reducing CI time for large documentation repos where most links remain unchanged between runs.

## Problem

A repo with 500 external links checks all 500 on every CI push, even if only one Markdown file changed. The in-memory cache (`linter.go:69`) helps within a run but provides no benefit across runs.

```
Run 1:  500 URLs → 500 HTTP requests → cache discarded on exit
Run 2:  500 URLs → 500 HTTP requests → same URLs re-checked
```

### Files to modify

- `internal/rule/external_link.go`
- `internal/linter/linter.go`
- `internal/config/config.go`

### Cache design

**Cache entry:**

```go
type cacheEntry struct {
    Status    int       `json:"status"`
    CachedAt  time.Time `json:"cached_at"`
}
```

**What to cache:** successful responses only (2xx and 3xx). Failed URLs are re-checked on every run to detect recovery. Temporary failures (e.g., transient 503) should not be persisted and silently hidden.

**TTL:** configurable, default 24 hours.

**Format:** JSON file, one entry per URL. Human-readable so users can inspect or manually invalidate entries.

```json
{
  "https://example.com/docs": { "status": 200, "cached_at": "2026-05-08T06:00:00Z" },
  "https://github.com/foo/bar": { "status": 301, "cached_at": "2026-05-08T06:00:00Z" }
}
```

### New config options

```json
{
  "rules": {
    "external-link": {
      "enabled": true,
      "cache_file": ".gomarklint-cache.json",
      "cache_ttl_hours": 24
    }
  }
}
```

`cache_file` defaults to empty string (cache disabled). Users opt in explicitly.

### Lifecycle

1. On startup, load cache file if it exists; populate `sync.Map` with non-expired entries
2. During the run, `sync.Map` works as today — HTTP results stored in memory
3. On exit, write successful entries back to the cache file (merge with existing, drop expired entries)

### CI usage (GitHub Actions example)

gomarklint writes and reads the file; caching across runs is handled by the CI system:

```yaml
- uses: actions/cache@v4
  with:
    path: .gomarklint-cache.json
    key: gomarklint-url-cache-${{ runner.os }}
```

gomarklint itself has no GitHub Actions dependency — the cache file is just a local file.

## Acceptance criteria

- [ ] Cache file is loaded on startup; entries within TTL are used without HTTP requests
- [ ] Expired entries are ignored and re-checked
- [ ] Only 2xx/3xx results are written to the cache file
- [ ] Cache file is written on clean exit (not on panic/signal)
- [ ] `cache_file` defaults to `""` (opt-in; no file written unless configured)
- [ ] `cache_ttl_hours` is configurable; default 24
- [ ] Unit tests for load/save, TTL expiry, and successful-only filtering
- [ ] Existing behavior is unchanged when `cache_file` is not set
